### PR TITLE
Add feedback in docs

### PR DESCRIPTION
**Pull request by ThyeeZz**

### CODE REVIEW

File: src/components/scoredFeedback/index.jsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+          feedback !== '' ? styles.hideOptions : ''
+        }`}
+      >
+        <span className={`${styles.text}`}>{feedbackText.text1}</span>
+        <span
+          className={`${styles.iconWrapper} ${styles.hoverLike}`}
+          role="button"
+          tabIndex={0}
+          onClick={() => handleFeedback('like')}
+          onKeyDown={() => handleFeedback('dislike')}
```
Comment (Tumao727) [inline src/components/scoredFeedback/index.jsx:75]: why click event param is `like` but keyDown event param is `dislike`

File: src/components/scoredFeedback/index.jsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+const ScoredFeedback = ({ old, feedbackText }) => {
+  const { pathname } = globalHistory.location;
+
+  const [feedback, setFeedback] = useState('');
+  const [isFeedback, setIsFeedback] = useState(true);
+
+  const handleUpdateStatisData = async (islike = false) => {
+    const { like, dislike, sha } = await readStatisData();
+    if (islike) {
```
Comment (Tumao727) [inline src/components/scoredFeedback/index.jsx:16]: `const pathnameCount = statistic[value][pathname]`
`const content = pathnameCount ? {...statistic, pathname: pathnameCount + 1} : {...statistic, pathname: 1}`

File: src/http/http.js [hunk 3]
```diff
@@ -53,4 +59,64 @@ export const sendQuestion = ({ quest = '', email = '' }) => {
...
+        repo: feedbackRepo,
+        path: 'statistics/index.json',
+      }
+    );
+    const { content, sha } = res.data;
+    const { like, dislike } = JSON.parse(
+      window.atob(content.replaceAll('\n', ''))
+    );
+    return {
+      like,
```
Comment (Tumao727) [inline src/http/http.js:85]: `statistic: JSON.parse(
      window.atob(content.replaceAll('\n', ''))
    );`

File: src/components/scoredFeedback/index.jsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        feedback: val,
+      });
+
+      setTimeout(() => {
+        setIsFeedback(true);
+      }, 3000);
+    }
+    window.localStorage.setItem(FEEDBACK_INFO, JSON.stringify(feedbackInfo));
+    setFeedback(val);
+    handleUpdateStatisData(Boolean(val === 'like'));
```
Comment (Tumao727) [inline src/components/scoredFeedback/index.jsx:46]: `handleUpdateStatisData(Boolean(val));`

File: src/components/scoredFeedback/index.jsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      message: 'update statistics data',
+    });
+  };
+
+  const handleFeedback = val => {
+    const feedbackInfoString = window.localStorage.getItem(FEEDBACK_INFO);
+    const feedbackInfo = feedbackInfoString
+      ? JSON.parse(feedbackInfoString)
+      : [];
+    if (!isFeedback) {
```
Comment (Tumao727) [inline src/components/scoredFeedback/index.jsx:34]: delete useless condition

File: src/components/scoredFeedback/index.jsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    window.localStorage.setItem(FEEDBACK_INFO, JSON.stringify(feedbackInfo));
+    setFeedback(val);
+    handleUpdateStatisData(Boolean(val === 'like'));
+  };
+
+  useEffect(() => {
+    // make sure whether this doc has been feedbacked
+    const feedbackInfoString = window.localStorage.getItem(FEEDBACK_INFO);
+    const feedbackInfo = feedbackInfoString
+      ? JSON.parse(feedbackInfoString)
```
Comment (nameczz) [inline src/components/scoredFeedback/index.jsx:53]: need try catch for Json.parse

File: src/components/scoredFeedback/index.jsx [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+  const [isFeedback, setIsFeedback] = useState(true);
+
+  const handleUpdateStatisData = async value => {
+    const { statistic, sha } = await readStatisData();
+    const info = statistic[value];
+
+    const pathnameCount = statistic[value][pathname];
+    const content = pathnameCount
+      ? { ...statistic, [value]: { ...info, [pathname]: pathnameCount + 1 } }
+      : { ...statistic, [value]: { ...info, [pathname]: 1 } };
```
Comment (Tumao727) [inline src/components/scoredFeedback/index.jsx:21]:
```js
const pathnameCount = pathname in info? info[pathname] + 1 : 1
const content = {...statistic, [value]: {...info, [pathname]: pathnameCount}}
```

File: src/components/scoredFeedback/index.jsx [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+      feedbackInfo.push({
+        md_id: old,
+        feedback: val,
+      });
+
+      setTimeout(() => {
+        setIsFeedback(true);
+      }, 3000);
+
+      // window.localStorage.setItem(FEEDBACK_INFO, JSON.stringify(feedbackInfo));
```
Comment (Tumao727) [inline src/components/scoredFeedback/index.jsx:46]: why hide this?

**Pull request merged**